hazelcast/src/main/java/com/hazelcast/concurrent/lock/operations/LockOperation.java

```diff
@@ -45,6 +46,12 @@ public LockOperation(ObjectNamespace namespace, Data key, long threadId, long le
 
     @Override
     public void run() throws Exception {
+        // if service is a LockInterceptorService, notify it a key is about to be locked
+        Object targetService = getNodeEngine().getService(namespace.getServiceName());
+        if (targetService instanceof LockInterceptorService) {
+            ((LockInterceptorService) targetService).onBeforeLock(namespace.getObjectName(), key);
```

Comment:
This code-block can also be put `LockBackupOperation`?

Reply:
+1, pulled as separate method to `AbstractLockOperation` and used in both `LockOperation` and `LockBackupOperation`.